src/components/graph/graph.helper.js

```diff
@@ -43,18 +44,36 @@ const NODE_PROPS_WHITELIST = ['id', 'highlighted', 'x', 'y', 'index', 'vy', 'vx'
  * @param  {number} width - the width of the container area of the graph.
  * @param  {number} height - the height of the container area of the graph.
  * @param  {number} gravity - the force strength applied to the graph.
+ * @param  {callback} onCollide - the method takes a node as its param and returns a collision force radius.
  * @returns {Object} returns the simulation instance to be consumed.
  * @memberof Graph/helper
  */
-function _createForceSimulation(width, height, gravity) {
+function _createForceSimulation(width, height, gravity, onCollide) {
     const frx = d3ForceX(width / 2).strength(CONST.FORCE_X);
     const fry = d3ForceY(height / 2).strength(CONST.FORCE_Y);
     const forceStrength = gravity;
+    const onCollision = _isValidCollisionFunction(onCollide) ? onCollide : d3ForceCollide;
 
     return d3ForceSimulation()
         .force('charge', d3ForceManyBody().strength(forceStrength))
         .force('x', frx)
-        .force('y', fry);
+        .force('y', fry)
+        .force(
+            'collide',
+            d3ForceCollide(node => {
+                onCollision(node);
+            })
+        );
+}
+
+/**
+ *
+ * @param {callback} onCollide - the method takes a node as its param and returns a collision force radius.
+ * @returns {*|boolean} value that indicates if onCollide is a valid collision function
+ * @private
+ */
+function _isValidCollisionFunction(onCollide) {
+    return onCollide && typeof onCollide === 'function';
```

Comment:
add `!!` so it always returns `boolean`:

`return !!onCollide && typeof onCollide === 'function';`